.github/workflows/bake-publish.yml

```
@@ -0,0 +1,128 @@
+---
+# This is a reusable workflow for building and publishing Docker images
+# using a bake file.
+
+on:
+  workflow_call:
+    inputs:
+      file:
+        description: 'The file name of the bake file to use. If not provided will use default.'
+        type: string
+        default: 'docker-bake.hcl'
+        required: false
+      group:
+        description: 'Group to use from the bake file.'
+        type: string
+        default: ''
+        required: true
+      registry:
+        description: 'The docker registry to use.'
```

Should we use `container` instead of `docker` to be clearer?
```suggestion
        description: 'The container registry to use.'
```